lib/ansible/module_utils/basic.py

```diff
@@ -708,6 +709,12 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self.no_log_values = set()
         # Use the argspec to determine which args are no_log
         for arg_name, arg_opts in self.argument_spec.items():
+            if (arg_opts.get('deprecated', False) or arg_opts.get('deprecated_version')) and arg_name in self.params:
```

Comment:
here you should not need to check for deprecated if we are going to mandate deprecated_version